Mathlib/CategoryTheory/Equivalence.lean

```diff
@@ -0,0 +1,816 @@
+  slice_lhs 2 3 => erw [e.unit.naturality]
+  slice_lhs 1 2 => erw [e.unit.naturality]
+  slice_lhs 4 4 =>
+    rw [← Iso.hom_inv_id_assoc (e.inverse.mapIso (e.counitIso.app _)) (e.unitInv.app _)]
+  slice_lhs 3 4 =>
+    erw [← map_comp e.inverse, e.counit.naturality]
+    erw [(e.counitIso.app _).hom_inv_id, map_id]
+  erw [id_comp]
+  slice_lhs 2 3 => erw [← map_comp e.inverse, e.counitIso.inv.naturality, map_comp]
+  slice_lhs 3 4 => erw [e.unitInv.naturality]
+  slice_lhs 4 5 => erw [← map_comp (e.functor ⋙ e.inverse), (e.unitIso.app _).hom_inv_id, map_id]
+  erw [id_comp]
+  slice_lhs 3 4 => erw [← e.unitInv.naturality]
+  slice_lhs 2 3 =>
+    erw [← map_comp e.inverse, ← e.counitIso.inv.naturality, (e.counitIso.app _).hom_inv_id,
+      map_id]
+  erw [id_comp, (e.unitIso.app _).hom_inv_id]; rfl
+#align category_theory.equivalence.unit_inverse_comp CategoryTheory.Equivalence.unit_inverse_comp
+
+@[simp]
+theorem inverse_counit_inv_comp (e : C ≌ D) (Y : D) :
+    e.inverse.map (e.counitInv.app Y) ≫ e.unitInv.app (e.inverse.obj Y) = 𝟙 (e.inverse.obj Y) :=
+  by
+  erw [Iso.inv_eq_inv (e.unitIso.app (e.inverse.obj Y) ≪≫ e.inverse.mapIso (e.counitIso.app Y))
+      (Iso.refl _)]
+  exact e.unit_inverse_comp Y
+#align category_theory.equivalence.inverse_counit_inv_comp 
+  CategoryTheory.Equivalence.inverse_counit_inv_comp
+
+theorem unit_app_inverse (e : C ≌ D) (Y : D) :
+    e.unit.app (e.inverse.obj Y) = e.inverse.map (e.counitInv.app Y) := by
+  erw [← Iso.comp_hom_eq_id (e.inverse.mapIso (e.counitIso.app Y)), unit_inverse_comp]
+  dsimp
+#align category_theory.equivalence.unit_app_inverse CategoryTheory.Equivalence.unit_app_inverse
+
+theorem unit_inv_app_inverse (e : C ≌ D) (Y : D) :
+    e.unitInv.app (e.inverse.obj Y) = e.inverse.map (e.counit.app Y) := by
+  symm
+  erw [← Iso.hom_comp_eq_id (e.unitIso.app _), unit_inverse_comp]
+  rfl
+#align
+  category_theory.equivalence.unit_inv_app_inverse CategoryTheory.Equivalence.unit_inv_app_inverse
+
+@[simp]
+theorem fun_inv_map (e : C ≌ D) (X Y : D) (f : X ⟶ Y) :
+    e.functor.map (e.inverse.map f) = e.counit.app X ≫ f ≫ e.counitInv.app Y :=
+  (NatIso.naturality_2 e.counitIso f).symm
+#align category_theory.equivalence.fun_inv_map CategoryTheory.Equivalence.fun_inv_map
+
+@[simp]
+theorem inv_fun_map (e : C ≌ D) (X Y : C) (f : X ⟶ Y) :
+    e.inverse.map (e.functor.map f) = e.unitInv.app X ≫ f ≫ e.unit.app Y :=
+  (NatIso.naturality_1 e.unitIso f).symm
+#align category_theory.equivalence.inv_fun_map CategoryTheory.Equivalence.inv_fun_map
+
+section
+
+-- In this section we convert an arbitrary equivalence to a half-adjoint equivalence.
+variable {F : C ⥤ D} {G : D ⥤ C} (η : 𝟭 C ≅ F ⋙ G) (ε : G ⋙ F ≅ 𝟭 D)
+
+/-- If `η : 𝟭 C ≅ F ⋙ G` is part of a (not necessarily half-adjoint) equivalence, we can upgrade it
+to a refined natural isomorphism `adjointify_η η : 𝟭 C ≅ F ⋙ G` which exhibits the properties
+required for a half-adjoint equivalence. See `equivalence.mk`. -/
```

Comment:
```suggestion
required for a half-adjoint equivalence. See `Equivalence.mk`. -/
```

I'm not sure what the new naming conventions say about lemma names such as `equivalence_mk'_foo`. Should these be changed to `Equivalence_mk'_foo`?

Reply:
You are right. I didn’t run the new comment naming convention fixer on this. So good might not be the only incident

Reply:
Okay I think I got all relevant ones now. There is a few theorems that have names along the lines of `of_equivalence` but take an `IsEquivalence` argument so I wasn't sure whether I want to mention the name of a struct there?

Reply:
I think that while the port is going on, we should keep the names as close as possible to their mathlib3 counterparts. So the names look good to me after your last change.